src/formatDate.js

```diff
@@ -8,7 +8,29 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  const newDate = [];
+  const dateSet = {};
+
+  for (let i = 0; i < 3; i++) {
+    dateSet[fromFormat[i]] = dateParts[i];
+  }
+
+  if (dateSet.YYYY) {
```

Comment:
This conditional check is not sufficient to determine if the year needs to be converted from YYYY to YY. It will always be true if the date includes a year, regardless of the desired format. You should check if the 'toFormat' array requires a YY instead of YYYY.